__Задача 1__

Коробки с шоколадом упаковываются автоматически. Их средняя масса равна 1.06 кг. Известно, что 5% коробок имеют массу, меньшую 1 кг. Найдите: а) среднее квадратическое отклонение массы коробки, б) процент коробок, имеющих массу больше 1.1 кг.

_Подсказка_. 1. Можно считать, что распределение массы коробки нормальное. 2. Найдите такое значение `scale`, для которого значение `cdf(x=1, loc=1.06, scale=scale)` близко к 0.05. Точности 0.0001 будет достаточно.

Пусть X - нормально распределенная случайная величина, равная массе коробки с конфетами, тогда a = 1,06 (мат ожидание), σ (среднее квадратичное отклонение)

Нужно найти scale, чтобы было верно одно из условий: (оно должно отклоняться от 0.05 не больше, чем на 0.0001)

* cdf(x = 1, loc = 1.06, scale = scale) < 0.05 - 0.0001

* или cdf(x = 1, loc = 1.06, scale = scale) > 0.05 + 0.0001

In [1]:
import numpy as np
from scipy import stats

In [2]:
loc = 1.06

In [22]:
start = 0
end = 1

target = 0.05
err = 0.0001

print('cdf \tscale')


while True:
    scale = (start + end)/2
    cdf = stats.norm.cdf(1, loc=loc, scale=scale)
    
    print(round(cdf, 4), '\t', round(scale, 4))
    
    if cdf <= target - err:
        start = scale
    elif cdf >= target + err:
        end = scale
    else:
        break
        
scale        

cdf 	scale
0.4522 	 0.5
0.4052 	 0.25
0.3156 	 0.125
0.1685 	 0.0625
0.0274 	 0.0312
0.1003 	 0.0469
0.0623 	 0.0391
0.0439 	 0.0352
0.053 	 0.0371
0.0484 	 0.0361
0.0507 	 0.0366
0.0495 	 0.0364
0.0501 	 0.0365
0.0498 	 0.0364
0.05 	 0.0365


0.036468505859375

Проверка:

In [4]:
stats.norm.cdf(1, loc=loc, scale=scale )

0.049958594078015874

Теперь можно найти вероятность того, что коробка будет иметьмассу более 1.1 кг. Воспользуемся функцией распределения:
 p = 1 - F(1.1)

In [5]:
1 - stats.norm.cdf(1.1, loc = loc, scale = scale)

0.1363563656878326

Вероятность 13.6 %

__Задача 2__

Коробка содержит 30 конфет. Известно, что масса каждой конфеты распределена равномерно в промежутке от 12 до 14 граммов. Используя центральную предельную теорему, найти вероятность, что масса всей коробки будет: а) меньше 390 граммов, б) больше 395 граммов, в) от 380 до 400 граммов.

Массой самой коробки можно пренебречь.

_Подсказка_. Для равномерного распределения в промежутке `[a, b]`:
1. мат. ожидание равно `(a + b) / 2`,
2. дисперсия равна `(b - a)^2 / 12`.

С помощью центральной предельной теоремы можно найти распределение массы всей коробки(30 конфет). Тогда мат ожидание и дисперсия массы каждой конфеты:


$$ m = \dfrac{{a} + b}{{2}} = 13,  b = \dfrac{({b} -a)^2}{12} = 10$$

По центральной теореме масса всей коробки имеет нормальное распределение со следующими параметрами:
$$\mu = 30 \cdot m = 390,    \sigma^2= 30 \cdot d = 10 $$

Теперь мы можем найти искомые вероятности:

In [7]:
mu = 390
sigma = np.sqrt(10)

norm = stats.norm(loc= mu, scale = sigma)

Вероятность, что коробка будет весить меньше 390 граммов:

In [8]:
norm.cdf(390)

0.5

Вероятность, что коробка будет весить больше 395 грамм:

In [9]:
1 - norm.cdf(395)

0.056923149003329065

Вероятность, что масса коробки будет в диапазоне от 380 до 400 граммов:


In [19]:
norm.cdf(400) - norm.cdf(380)

0.9984345977419975